app/uk/gov/hmrc/agentuserclientdetails/controllers/ClientListController.scala

```diff
@@ -58,36 +58,50 @@ class ClientListController @Inject() (
       }
     }
 
-  def getPaginatedClients(arn: Arn, page: Int = 1, pageSize: Int = 20): Action[AnyContent] = Action.async {
-    implicit request =>
-      withAuthorisedAgent(allowStandardUser = true) { _ =>
-        withGroupIdFor(arn) { groupId =>
-          espConnector
-            .getClientsForGroupId(groupId)
-            .map { clients =>
-              val pageStart = (page - 1) * pageSize
-              val pageEnd = pageStart + pageSize
-              val currentPageContent = clients.slice(pageStart, Math.min(pageEnd, clients.length - 1))
-              val numberOfPages = Math.ceil(clients.length.toDouble / pageSize.toDouble).toInt
-              Ok(
-                Json.toJson(
-                  PaginatedList[Client](
-                    pageContent = currentPageContent,
-                    paginationMetaData = PaginationMetaData(
-                      firstPage = page == 1,
-                      lastPage = numberOfPages == page,
-                      totalSize = clients.length,
-                      pageSize = pageSize,
-                      totalPages = numberOfPages,
-                      currentPageNumber = page,
-                      currentPageSize = currentPageContent.length
-                    )
+  def getPaginatedClients(
+    arn: Arn,
+    page: Int = 1,
+    pageSize: Int = 20,
+    search: Option[String] = None,
+    filter: Option[String] = None
+  ): Action[AnyContent] = Action.async { implicit request =>
+    withAuthorisedAgent(allowStandardUser = true) { _ =>
+      withGroupIdFor(arn) { groupId =>
+        espConnector
+          .getClientsForGroupId(groupId)
+          .map { clients =>
+            val clientsMatchingSearch = search.fold(clients) { searchTerm =>
+              clients.filter(c => c.friendlyName.toLowerCase.contains(searchTerm.toLowerCase))
```

Comment:
search should also search tax reference, but it'll okay for what we need

Reply:
yeah for now